plugins/mac-capture/mac-sck-audio-capture.m

```diff
@@ -0,0 +1,385 @@
+				  NSError *_Nullable error) {
+			if (error && error.code != 3808) {
+				MACCAP_ERR(
+					"destroy_audio_screen_stream: Failed to stop stream with error %s\n",
+					[[error localizedFailureReason]
+						cStringUsingEncoding:
+							NSUTF8StringEncoding]);
+			}
+			os_event_signal(sc->disp_finished);
+		}];
+		os_event_wait(sc->disp_finished);
+	}
+
+	if (sc->stream_properties) {
+		[sc->stream_properties release];
+		sc->stream_properties = NULL;
+	}
+
+	if (sc->disp) {
+		[sc->disp release];
+		sc->disp = NULL;
+	}
+
+	os_event_destroy(sc->disp_finished);
+	os_event_destroy(sc->stream_start_completed);
+}
+
+static void sck_audio_capture_destroy(void *data)
+{
+	struct screen_capture *sc = data;
+
+	if (!sc)
+		return;
+
+	obs_enter_graphics();
+
+	destroy_audio_screen_stream(sc);
+
+	obs_leave_graphics();
+
+	if (sc->shareable_content) {
+		os_sem_wait(sc->shareable_content_available);
+		[sc->shareable_content release];
+		os_sem_destroy(sc->shareable_content_available);
+		sc->shareable_content_available = NULL;
+	}
+
+	if (sc->capture_delegate) {
+		[sc->capture_delegate release];
+	}
+	[sc->application_id release];
+
+	pthread_mutex_destroy(&sc->mutex);
+	bfree(sc);
+}
+
+static bool init_audio_screen_stream(struct screen_capture *sc)
+{
+	SCContentFilter *content_filter;
+
+	sc->stream_properties = [[SCStreamConfiguration alloc] init];
+	os_sem_wait(sc->shareable_content_available);
+
+	SCDisplay * (^get_target_display)() = ^SCDisplay *()
+	{
+		__block SCDisplay *target_display = nil;
+		[sc->shareable_content.displays
+			indexOfObjectPassingTest:^BOOL(
+				SCDisplay *_Nonnull display, NSUInteger idx,
+				BOOL *_Nonnull stop) {
+				if (display.displayID == sc->display) {
+					target_display = sc->shareable_content
+								 .displays[idx];
+					*stop = TRUE;
+				}
+				return *stop;
+			}];
+		return target_display;
+	};
+
+	switch (sc->capture_type) {
+	case ScreenCaptureAudioDesktopStream: {
+		SCDisplay *target_display = get_target_display();
+
+		content_filter = [[SCContentFilter alloc]
+			 initWithDisplay:target_display
+			excludingWindows:[[NSArray alloc] init]];
+	} break;
+	case ScreenCaptureAudioApplicationStream: {
+		SCDisplay *target_display = get_target_display();
+		__block SCRunningApplication *target_application = nil;
+		{
+			[sc->shareable_content.applications
+				indexOfObjectPassingTest:^BOOL(
+					SCRunningApplication
+						*_Nonnull application,
+					NSUInteger idx, BOOL *_Nonnull stop) {
+					if ([application.bundleIdentifier
+						    isEqualToString:
+							    sc->
+							    application_id]) {
+						target_application =
+							sc->shareable_content
+								.applications
+									[idx];
+						*stop = TRUE;
+					}
+					return *stop;
+				}];
+		}
+		NSArray *target_application_array = [[NSArray alloc]
+			initWithObjects:target_application, nil];
+
+		content_filter = [[SCContentFilter alloc]
+			      initWithDisplay:target_display
```

Comment:
`target_display` and `target_application_array` need to manually released, otherwise they leak memory.

Reply:
Fixed to release ``target_application_array``. I don't see an alloc call for ``target_display`` though, and it seems to come from ``sc->shareable_content.displays[index]``, so I did not add a ``release`` for it. I will double check it later, currently having issues with the Xcode leaks instrument which I think will require a macOS or Xcode update.

Reply:
You can just do an analytical build (cmd+shift+b) to have it check for these kind of leaks.

Reply:
Alright, analysis seems clear for leaks, got the leaks Instrument working as well and ran some tests and things seem clear there as well, no warnings about ``target_display``.